tmt/cli.py

```diff
@@ -1865,14 +1865,28 @@ def status(
 #  Clean
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+# Supported clean resources
+CLEAN_RESOURCES: list[str] = ["guests", "runs", "images"]
 
-# ignore[arg-type]: click code expects click.Context, but we use our own type for better type
-# inference. See Context and ContextObjects above.
-@main.group(chain=True, invoke_without_command=True, cls=CustomGroup)  # type: ignore[arg-type]
+
+@main.group(chain=True, invoke_without_command=True, cls=CustomGroup)
 @pass_context
+@option(
+    '-l', '--last', is_flag=True,
+    help='Clean resources related to the last run.')
+@option(
+    '-i', '--id', 'id_', metavar="ID", multiple=True,
+    help='Identifier (name or directory path) of the run to be cleaned.')
+@option(
+    '-s', '--skip', choices=CLEAN_RESOURCES,
+    help='The resources which should be kept on the disk.', multiple=True)
 @verbosity_options
 @dry_options
-def clean(context: Context, **kwargs: Any) -> None:
+def clean(context: Context,
+          last: bool,
+          id_: tuple[str, ...],
```

Comment:
Does this actually make any difference?  
After looking at the clean code, my earlier comment about matching `N` to `run-N` paths is apparently related to this PR.

Reply:
Just copy-pasted it from another subcommand.

Reply:
If can/should be simplified let's do them all at once in a separate patch.

Reply:
`tuple[str]` is only one string item while `tuple[str, …]` sllows that, but alo none or seven string items. It’s not the same as `list`.

Reply:
Thanks for the clarification, @happz!